onnxruntime/contrib_ops/cuda/custom_layer_norm.cu

```diff
@@ -0,0 +1,322 @@
+/**
+* Copyright (c) 2016-present, Facebook, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+//
+// Copyright (c) 2017, NVIDIA CORPORATION. All rights reserved.
+// NVIDIA/apex is licensed under the
+// BSD 3 - Clause "New" or "Revised" License
+//
+
+/* Modifications Copyright (c) Microsoft. */
+
+#include "core/providers/cuda/cu_inc/common.cuh"
+#include <cooperative_groups.h>
+#include "custom_layer_norm.cuh"
+
+namespace onnxruntime {
+namespace contrib {
+namespace cuda {
+
+namespace cg = cooperative_groups;
+using namespace onnxruntime::cuda;
+/*
+Custom kernel for Layer Normalization, adapted from https://github.com/microsoft/DeepSpeed.
+This layer requires template parameters since it relies fairly heavily on unrolling loops.
+The kernel is invoked when the n2:the number of elements is a multiple of a power of 2. 
```

Comment:
>the n2:the number [](start = 27, length = 17)

"the n2:the number"
should this be "the number"?

Reply:
edited for clarification. This indicates the 'n2' argument in the kernel launch function.